docs/plugins/private-plugins.md

```diff
@@ -0,0 +1,268 @@
+---
+id: private-plugins
+slug: /private-plugins
+toc_max_heading_level: 2
+---
+
+# Creating and Using Private Plugins
+
+Private Plugins enable you to create [custom plugins](/plugins/overview/) exclusively for your organization's use. These plugins let you build secure, proprietary integrations while protecting sensitive business logic.
+
+:::note
+Private Plugins are only **available for Enterprise Plan users**.
+:::
+
+Unlike community plugins hosted on public package managers, private plugins reside in your own Git repository, giving you complete control over your plugin infrastructure and code.
+
+## Create Your Private Plugins Repository
+
+First, set up a dedicated Git repository to host your plugin's source code and builds:
+
+1. Create a new Git repository specifically for your private plugins
+2. Create a `plugins` folder in the repository root
+
+The `plugins` folder will contain separate folders for each of your private plugins.
+
+![Private Plugins Repo](./assets/private-plugins/private-plugins-repo.png)
+
+## Configure Git Settings In Platform Console
+
+Next, configure your private plugins repository in the Platform Console:
+
+1. Navigate to your project's **Platform Console** from the top header
+2. Click the **Private Plugins** tab
+3. Select **Git Settings** in the left sidebar
+4. Enable the **Override default settings** toggle
+5. Choose your dedicated plugin repository from the dropdown
+
+:::important
+Never use the same repository for private plugins and generated services to avoid build conflicts.
+:::
+
+![Publish Your Platform Console Changes](./assets/private-plugins/configure-git-settings.png)
+
+## Add a New Private Plugin
+
+Next, add a new private plugin to your Platform Console:
+
+1. In the Private Plugins tab's sidebar, click the `Add Private Plugin` field
+2. Enter a unique name using kebab-case (e.g., `acme-corp-authentication`)
+3. Set the plugin's **Display Name** and **Description** - these will appear in the UI when selecting plugins
+4. Choose the appropriate code generator (Node.js or .NET) that matches your target services
+
+The Plugin Id you enter will become the name of your plugin's folder within the `plugins` folder.
```

Comment:
"will become" sounds like we will create this folder. Maybe we need to say that the ID should match the folder name of the plugin in the repo or such